src/cloud-api-adaptor/pkg/securecomms/kubemgr/kubemgr.go

```diff
@@ -0,0 +1,127 @@
+package kubemgr
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"flag"
+	"fmt"
+	"log"
+	"path/filepath"
+
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/securecomms/sshutil"
+	"golang.org/x/crypto/ssh"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+var KubeMgr *KubeMgrStruct
+
+const (
+	cocoNamespace = "confidential-containers-system"
+)
+
+type KubeMgrStruct struct {
+	client        *kubernetes.Clientset
+	cocoNamespace string
+}
+
+var SkipVerify bool
+
+func InitKubeMgr() error {
+	var err error
+	KubeMgr = &KubeMgrStruct{
+		cocoNamespace: cocoNamespace,
+	}
+
+	var kubeCfg *rest.Config
+	var devKubeConfigStr *string
+
+	// Try to detect in-cluster config
+	if kubeCfg, err = rest.InClusterConfig(); err != nil {
+		// Not running in cluster
+		if home := homedir.HomeDir(); home != "" {
+			devKubeConfigStr = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
```

Comment:
Isn't it customary to also check the `KUBECONFIG` environment variable?

Reply:
The customary way as far as I can say is the one implemented. This code is for dev and testing only.

Reply:
I agree with Christophe. I ran this code and it failed with:
```
go run ./test/securecomms/double/main.go
go: downloading github.com/vishvananda/netns v0.0.0-20210104183010-2eb08e3e575f
go: downloading google.golang.org/protobuf v1.31.0
failed to initialize KubeMgr: no Config found to access KubeApi! err: stat /root/.kube/config: no such file or directory
```
whereas I have KUBECONFIG defined, so fully expected it to pick up that and can't see a reason to not check that?

Reply:
NP, 
adding KUBECONFIG, 
trying each file in KUBECONFIG before using flags

```golang
func getKubeConfigInVitro() (*rest.Config, error) {
	...
	// Not running in cluster
	kubeConfigEnv := os.Getenv("KUBECONFIG")
	envVarFilePaths := filepath.SplitList(kubeConfigEnv)
	for _, envVarFilePath := range envVarFilePaths {
		if kubeCfg, err = clientcmd.BuildConfigFromFlags("", envVarFilePath); err == nil {
			return kubeCfg, nil
		}
	}
        ... //try flags and the default ~/.kube/config
```